db/fdb_fend.c

```diff
@@ -964,30 +964,30 @@ static int check_table_fdb(fdb_t *fdb, fdb_tbl_t *tbl, int initial,
     /* prepackaged select */
     if (versioned) {
         if (initial) {
-            snprintf(sql, sizeof(sql),
+            sql = sqlite3_mprintf(
                      "select *, table_version(tbl_name) from sqlite_master"
-                     " where tbl_name='%s' collate nocase or tbl_name="
+                     " where tbl_name='%q' collate nocase or tbl_name="
                      "'sqlite_stat1' or "
                      "tbl_name='sqlite_stat4'",
                      tbl->name);
         } else {
-            snprintf(sql, sizeof(sql),
+            sql = sqlite3_mprintf(
                      "select *, table_version(tbl_name) from sqlite_master"
-                     " where tbl_name='%s' collate nocase",
+                     " where tbl_name='%q' collate nocase",
                      tbl->name);
         }
     } else {
         /* fallback to old un-versioned implementation */
         if (initial) {
-            snprintf(sql, sizeof(sql),
+            sql = sqlite3_mprintf(
                      "select * from sqlite_master"
-                     " where tbl_name='%s' or tbl_name='sqlite_stat1' or "
+                     " where tbl_name='%q' or tbl_name='sqlite_stat1' or "
                      "tbl_name='sqlite_stat4' collate nocase",
                      tbl->name);
         } else {
-            snprintf(sql, sizeof(sql),
+            sql = sqlite3_mprintf(
                      "select * from sqlite_master"
-                     " where tbl_name='%s' collate nocase",
```

Comment:
We end up leaking sql if we hit one of the `goto run` cases below (eg: ending up on a node with a lower FDB version)